### PR TITLE
Fix #8507: Consider UnknownSymbols values for import/export purposes

### LINKED ISSUE #8507: export default for unresolved import gets elided
For the below code:

```javascript
import {Subscription} from 'rxjs/Subscription';
class Foo {
    sayHi(name) {
        return `Hi ${name}, welcome!`;
    }
}
export let sqr = x => x * x;
export default Subscription;
```

If `rxjs/Subscription' import does not resolve, then the emit is:

```javascript
"use strict";
var Foo = (function () {
    function Foo() {
    }
    Foo.prototype.sayHi = function (name) {
        return "Hi " + name + ", welcome!";
    };
    return Foo;
}());
exports.sqr = function (x) { return x * x; };
```

Notice that the `import` and `export default` has been elided.

This can be resolved by setting `isolatedModules`, or by referencing a property on Subscription (even if that could be a type on a namespace also - e.g. `export default Subscription.IFoo`).

This is especially an issue for Salsa when the input is JavaScript rather than TypeScript, as there are no errors/warning for unresolved modules/types (and the reference can't be to a type).

Spoke with Mohamed, and he agrees this should be fixed (and should be easy).

### PR DESCRIPTION
**Pull request by mhegazy**
Fixes #8507

This is important for JS files. regardless if we found the module or not, we should be always emitting the import/export to grantee valid transpilation. Also note that this is the same behavior for `--isolatedModules`, and it is currently an error case anyways, so there should not be any breaks from changing this behavior to working code.

### CODE REVIEW

**Comment by mhegazy** [comment]
//cc: @vladima

File: tests/baselines/reference/declarationEmit_UnknownImport.errors.txt [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 
 
-==== tests/cases/compiler/declarationEmit_UnknownImport.ts (3 errors) ====
+==== tests/cases/compiler/declarationEmit_UnknownImport.ts (4 errors) ====
     
     import Foo = SomeNonExistingName
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 !!! error TS2304: Cannot find name 'SomeNonExistingName'.
                  ~~~~~~~~~~~~~~~~~~~
+!!! error TS2304: Cannot find name 'SomeNonExistingName'.
```
Comment (vladima) [inline tests/baselines/reference/declarationEmit_UnknownImport.errors.txt]: extra `cannot find name 'name'` errors look annoying and not very useful, can we do something to suppress them?

Comment (mhegazy) [inline tests/baselines/reference/declarationEmit_UnknownImport.errors.txt]: @vladima, i removed the older `Cannot find name` error, as it was generated during declaration emit, and was not really needed. the new one however, is produced as we now call checkExpression to make sure we mark the identifier. i do not think there is an easy way to do that, unless we wire in a flag into checkExpression functions to not report errors. i do not think it is worth the complexity though.

**Comment by mhegazy** [comment]
@vladima any more comments?

**Pull request merged**